docs/develop/python/message-passing.mdx

```diff
@@ -221,132 +247,214 @@ async def main():
+@workflow.defn
+class GreetingWorkflow:
+    ...
+    @workflow.update
+    async def set_language(self, language: Language) -> Language:
+        if language not in self.greetings:
+            greeting = await workflow.execute_activity(
+                call_greeting_service,
+                language,
+                start_to_close_timeout=timedelta(seconds=10),
+            )
+            if greeting is None:
+                # An update validator cannot be async, so cannot be used to check that the remote
+                # call_greeting_service supports the requested language. Raising ApplicationError
+                # will fail the Update, but the WorkflowExecutionUpdateAccepted event will still be
+                # added to history.
+                raise ApplicationError(
+                    f"Greeting service does not support {language.name}"
+                )
+            self.greetings[language] = greeting
+        previous_language, self.language = self.language, language
+        return previous_language
+```
 
-- **Decorator Usage:** Apply `@workflow.update` to the method intended to handle updates.
-- **Overriding:** If a method with this decorator is overridden, the overriding method should also be decorated with `@workflow.update`.
-- **Validator Method:** Optionally, you can define a validator method for the update handler. This validator is specified using `@update_handler_method_name.validator` and is invoked before the update handler.
-- **Method Parameters:** Update handlers should only use positional parameters. For non-dynamic methods, it's recommended to use a single parameter that is an object or data class, which allows for future expansion of fields.
-- **Return Values:** The update handler can return a serializable value. This value is sent back to the caller of the update.
 
+#### Waiting
+[`await workflow.wait_condition(...)`](https://python.temporal.io/temporalio.workflow.html#wait_condition) is often useful in Temporal Workflow Definitions.
+You can use it in a handler as long as it's an `async def`:
 ```python
-# ...
     @workflow.update
-    async def update_workflow_status(self) -> str:
-        self.is_complete = True
-        return "Workflow status updated"
+    async def my_update(self, update_input: UpdateInput) -> str:
+        await workflow.wait_condition(
+            lambda: self.ready_for_update_to_execute(update_input)
+        )
 ```
 
-### Send an Update from a Temporal Client {#send-update-from-client}
 
-**How to send an Update from a Temporal Client**
+#### Use `asyncio.Lock` to prevent concurrent handler execution {#control-handler-concurrency}
+Sometimes you may write code that is incorrect if multiple instances of a handler are in progress concurrently.
```

Comment:
Plz include an uplink in each of these sections, and consider whether to reduce the amount of prose instead.